tests/tpubkey

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash -e
+# Copyright (C) 2022 Jakub Jelen <jjelen@redhat.com>
+# SPDX-License-Identifier: Apache-2.0
+
+source ${TESTSSRCDIR}/helpers.sh
+
+TSTOUT="${TMPPDIR}/testout"
+
+title PARA "Export RSA Public key to a file"
+ossl 'pkey -in $BASEURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export Public key to a file (pub-uri)"
+ossl 'pkey -in $PUBURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export Public key to a file (with pin)"
+ossl 'pkey -in $BASEURIWITHPIN -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Print Public key from private"
+ossl 'pkey -in $PRIURI -pubout -text' output
+FAIL=0
+echo $output | grep "PKCS11 RSA Public Key (2048 bits)" > /dev/null 2>&1 || FAIL=1
+if [ $FAIL -eq 1 ]; then
+    echo "Pkcs11 encoder function failed"
+    echo
+    echo "Original command output:"
+    echo "$output"
+    echo
+    exit 1
+fi
+
+title PARA "Export Public check error"
+FAIL=0
+ossl 'pkey -in pkcs11:id=%de%ad -pubin
+           -pubout -out ${TSTOUT}-invlid.pub' || FAIL=1
+if [ $FAIL -eq 0 ]; then
+    echo "Invalid pkcs11 uri resulted in no error exporting key"
+    exit 1
+fi
+
+title PARA "Export EC Public key to a file"
+ossl 'pkey -in $ECBASEURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export EC Public key to a file (pub-uri)"
+ossl 'pkey -in $ECPUBURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export EC Public key to a file (with pin)"
+ossl 'pkey -in $ECBASEURIWITHPIN -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Print EC Public key from private"
+ossl 'pkey -in $ECPRIURI -pubout -text' output
+FAIL=0
+echo $output | grep "PKCS11 EC Public Key (256 bits)" > /dev/null 2>&1 || FAIL=1
+if [ $FAIL -eq 1 ]; then
+    echo "Pkcs11 encoder function failed"
+    echo
+    echo "Original command output:"
+    echo "$output"
+    echo
+    exit 1
+fi
+
+# The softokn does not support removing only public key from the key pair
+if [ ! -z "$BASE2URI" ]; then
+    title PARA "Check we can get public keys from certificate objects"
+
+    title LINE "Export Public key to a file (priv-uri)"
+    ossl 'pkey -in $PRI2URI -pubin -pubout -out ${TSTOUT}-priv-cert.pub'
+    title LINE "Export Public key to a file (crt-uri)"
+    ossl 'pkey -in $CRT2URI -pubin -pubout -out ${TSTOUT}-cert.pub'
```

Comment:
This is not currently possible I think, as a cert is not a key, and openssl pkey handles only keys.

Reply:
Yeah. And I think this is the use case we are interested in, which would require some conversion. The certificate itself contains the public key, so the provider should be able to pull this information and serve it to the calling application if the separate public key is missing int he token.

Reply:
No, it is the other one we are interested in (from private), as we are testing the fallback code to extract public from private.